src/main/java/org/spongepowered/api/util/reporting/PrettyPrinter.java

```diff
@@ -0,0 +1,468 @@
+package org.spongepowered.api.util.reporting;
+
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.Logger;
+
+import java.io.PrintStream;
+import java.util.Collection;
+import java.util.Map;
+
+/**
+ * Prints information in a pretty box
+ */
+public interface PrettyPrinter {
```

Comment:
We're trying to *avoid* having more implementation in the API. It's the overall direction to move away from having this sort of implementation in the API when possible.